### PR TITLE
Sort by Bundle chained composition SPs

### LINKED ISSUE #5784: Support sorting by composition SPs
#### Description of Enhancement

We would like to be able to sort by fields within the resources inside their document bundles, e.g. composition.medicationadministration.whenPrepared.

Hapi currently throws an error on this:

    HTTP 400 Bad Request: HAPI-2286: Invalid _sort expression, can not chain more than once in a sort expression: composition.medicationadministration.effective-time=2023-12-13

### PR DESCRIPTION
**Pull request by michaelabuckley**
Fixes #5784

### CODE REVIEW

**Review by jamesagnew — APPROVED** [review]

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/ChainingR4SearchTest.java [hunk 1]
```diff
@@ -1643,36 +1642,22 @@ private void createDocumentBundleWithPatientDetails(String theBirthDate, String
 		bundle.addEntry().setResource(composition);
 		composition.getSubject().setReference(patient.getIdElement().getValue());
 		bundle.addEntry().setResource(patient);
-		myBundleDao.create(bundle, mySrd);
+		DaoMethodOutcome daoMethodOutcome = myBundleDao.create(bundle, mySrd);
```
Comment (jmarchionatto) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/ChainingR4SearchTest.java:1645]: nitpick: unused var

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SortTest.java [hunk 2]
```diff
@@ -348,20 +353,62 @@ public void testSortWithChainedSearch() {
...
+			);
+	}
+
+	/**
+	 * Define a composition SP for document Bundles, and sort by it.
+	 * The chain is referencing the Bundle contents.
+	 * @see https://smilecdr.com/docs/fhir_storage_relational/chained_searches_and_sorts.html#document-and-message-search-parameters
+	 */
+	@Test
+	void testSortByCompositionSP() {
```
Comment (jmarchionatto) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SortTest.java:371]: Would be nice to have a test for reverse order, as it is specifically mentioned in the issue

**Review by jmarchionatto — APPROVED** [review]
Looks good! Approved with a comment

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SortTest.java (hunk 2 above)
Comment (michaelabuckley) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SortTest.java:372]: good suggestion.

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/ChainingR4SearchTest.java (hunk 1 above)
Comment (michaelabuckley) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/dao/r4/ChainingR4SearchTest.java]: done

**Pull request merged**